src/tests/sqlsmith/src/sqlreduce/passes/mod.rs

```diff
@@ -71,7 +71,7 @@ pub trait Transform: Send + Sync {
     ///
     /// # Returns
     /// - The modified AST.
-    fn apply_on(&self, ast: &mut Ast, reduction_points: Vec<usize>) -> Ast;
+    fn apply_on(&self, ast: Ast, reduction_points: Vec<usize>) -> Ast;
```

Comment:
```suggestion
    fn apply_on(&self, ast: Ast, reduction_points: &[usize]) -> Ast;
```

We can even pass a slice of `reduction_points` to avoid the unnecessary `.to_vec` in the caller.

Reply:
Got it! I’ve fixed it in the new commit.